src/main/java/com/bushidowallet/core/bitcoin/tx/Transaction.java

```diff
@@ -1,356 +1,395 @@
+package com.bushidowallet.core.bitcoin.tx;
+
+import com.bushidowallet.core.bitcoin.Address;
+import com.bushidowallet.core.bitcoin.bip32.ECKey;
+import com.bushidowallet.core.bitcoin.bip32.Hash;
+import com.bushidowallet.core.bitcoin.script.Script;
+import com.bushidowallet.core.bitcoin.tx.input.Input;
+import com.bushidowallet.core.bitcoin.tx.input.MultiSigScriptHashInput;
+import com.bushidowallet.core.bitcoin.tx.input.PublicKeyHashInput;
+import com.bushidowallet.core.bitcoin.tx.output.Output;
+import com.bushidowallet.core.bitcoin.util.ByteReader;
+import com.bushidowallet.core.bitcoin.util.ByteWriter;
+import com.bushidowallet.core.crypto.util.ByteUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Jesion on 2015-03-31.
+ */
+public class Transaction {
+
+    private static int CURRENT_VERSION = 1;
+    private static int DEFAULT_NLOCKTIME = 0;
+    private static int MAXIMUM_EXTRA_SIZE = 4 + 9 + 9 + 4;
+    private static int CHANGE_OUTPUT_MAX_SIZE = 20 + 4 + 34 + 4;
+    private static long FEE_PER_KB = 10000;
+
+    public int version;
+
+    public List<Input> inputs;
+
+    public List<Output> outputs;
+
+    public int nLockTime;
+
+    private Script changeScript;
+
+    private int changeIndex;
+
+    private long fee;
+
+    public Transaction() {
+
+        inputs = new ArrayList<Input>();
+        outputs = new ArrayList<Output>();
+
+        version = CURRENT_VERSION;
+        nLockTime = DEFAULT_NLOCKTIME;
+
+        changeIndex = -1;
+        fee = -1;
+    }
+
+    public void sign(List<ECKey> keys) throws Exception {
+        sign(keys, TransactionSignature.SIGHASH_ALL);
+    }
+
+    /**
+     * Signs this Tx with a collection of private keys provided
+     *
+     * @param keys - collection of private keys used for signing
+     * @param sigType - see Signature
+     * @throws Exception
+     */
+    public void sign(List<ECKey> keys, int sigType) throws Exception {
+        for (int i = 0; i < keys.size(); i++) {
+            sign(keys.get(i), sigType);
+        }
+    }
+
+	public void sign(ECKey key, int sigType) throws Exception {
+		if (hasAllUtxoInfo()) {
+			List<TransactionSignature> signatures = getSignatures(key, sigType);
+			for (int j = 0; j < signatures.size(); j++) {
+				applySignature(signatures.get(j));
+			}
+		} else {
+			throw new Exception("Inputs not complete");
+		}
+	}
+
+	/**
+	 * add by Liwl
+	 * muilt sign by one person.
+	 * 
+	 * @param key
+	 * @param sigType
+	 * @return
+	 * @throws Exception
+	 */
+	public List<TransactionSignature> signByOne(ECKey key, int sigType) throws Exception {
+		if (hasAllUtxoInfo()) {
+			return getSignatures(key, sigType);
+		} else {
+			throw new Exception("Inputs not complete");
+		}
+	}
+
+	/**
+	 * add by Liwl
+	 * appley signature by one person.
+	 * 
+	 * @param signatures
+	 * @throws Exception
+	 */
+	public void applySignatureByOne(List<TransactionSignature> signatures) throws Exception {

```

Comment:
in a muilt-sign transaction,everyone can sign raw data whitout know other's private key.And should be no order.
For example,A,B,C in 2-3 tranction,B want to launch a deal and invite C to complet the transaction.C do not need to know B's private key,just sign the raw data and generate a signture data.C sign the data and add two signture data in transaction.
When add signture,should not order by pubkey in addSignature() method,it will occur exception when B add signture data in positon 1 because the list's size is 0!
Forgive me for my poor English.